packages/next/build/jest/jest.ts

```diff
@@ -151,8 +154,10 @@ export default function nextJest(options: { dir?: string } = {}) {
         },
 
         transformIgnorePatterns: [
-          // To match Next.js behavior node_modules is not transformed
-          '/node_modules/',
+          // To match Next.js behavior node_modules is not transformed, only `transpiledPackages`
+          `/node_modules/(?!(${transpiled}))/`,
+          `/node_modules/.pnpm/(?!(${transpiled.replace(/\//g, '\\+')})@)`,
+
```

Comment:
```suggestion
           ...(transpiled ? [
          `/node_modules/(?!(${transpiled}))/`,
          `/node_modules/.pnpm/(?!(${transpiled.replace(/\//g, '\\+')})@)`,
          ] : ['/node_modules/'])
```

Reply:
`transpiled` is always defined, this should be unnecessary.

Reply:
Nevermind, it can be `""` which is falsy. :+1: